### PR TITLE
refactor: Migrate saveModalActions to TypeScript

### PR DESCRIPTION
**Pull request by EnxDev**
### SUMMARY
Migrate saveModalActions to TypeScript

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N.A

### TESTING INSTRUCTIONS
- All tests should pass 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/actions/saveModalActions.ts [hunk 1]
```diff
@@ -17,126 +17,174 @@
...
+    }
+  });
+  return result;
+}
 
-const hasTemporalRangeFilter = formData =>
+const hasTemporalRangeFilter = (formData: Partial<FormData>): boolean =>
   (formData?.adhoc_filters || []).some(
-    filter => filter.operator === Operators.TemporalRange,
+    (filter: any) => filter.operator === Operators.TemporalRange,
```
Comment (geido) [inline superset-frontend/src/explore/actions/saveModalActions.ts:109]: We need to get rid of `any`s wherever possible

**Review by geido — APPROVED** [review]

File: superset-frontend/src/explore/actions/saveModalActions.ts [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+      return response.json.result.dashboards.map(
+        ({ id }: { id: number }) => id,
+      );
+    } catch (error) {
+      dispatch(saveSliceFailed());
+      throw error;
+    }
+  };
+export { QueryFormData };
```
Comment (geido) [inline superset-frontend/src/explore/actions/saveModalActions.ts:321]: @EnxDev this is creating some warning. Do we need it?

Comment (EnxDev) [reply superset-frontend/src/explore/actions/saveModalActions.ts:321]: I removed it here https://github.com/apache/superset/pull/28786

**Pull request merged**